src/gl/vcTextureHelper.cpp

```diff
@@ -61,3 +61,58 @@ udResult vcTexture_AsyncCreateFromFilename(vcTexture **ppTexture, udWorkerPool *
 epilogue:
   return result;
 }
+
+udResult vcTexture_ResizePixels(const void* pPixels, uint32_t width, uint32_t height, uint32_t targetSize, const void** ppResultPixels, uint32_t* pResultWidth, uint32_t* pResultHeight)
+{
+  // TODO: Add ability to upscale
+  udResult result;
+
+  const void* pLastPixels = pPixels;
+  uint32_t lastWidth = width;
+  uint32_t lastHeight = height;
+  int passCount = 0;
+
+  uint32_t currentSize = udMax(width, height);
+  UD_ERROR_IF(currentSize <= targetSize, udR_Failure_);
+
+  while ((currentSize >> passCount) > targetSize)
```

Comment:
Could this be simplified and drawn out of the iteration by something along the lines of totalPassCount = log2(currentSize) - log2(targetSize)....